### ISSUE TITLE
[Android only] Using WebBrowser in Expo Bare Workflow results in ExponentWebBrowser.dismissBrowser() being missing.

### ISSUE THREAD

**Issue by kitty7756** [expo/expo#4353]
### Environment

Expo CLI 2.18.2 environment info:
    System:
      OS: macOS High Sierra 10.13.6
      Shell: 5.0.7 - /usr/local/bin/bash
    Binaries:
      Node: 8.11.1 - ~/.nvm/versions/node/v8.11.1/bin/node
      Yarn: 1.7.0 - ~/.nvm/versions/node/v8.11.1/bin/yarn
      npm: 6.9.0 - ~/.nvm/versions/node/v8.11.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 10.0/10A255 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.3 => 16.6.3 
      react-native: 0.58.6 => 0.58.6 
      react-navigation: ^3.11.0 => 3.11.0 
    npmGlobalPackages:
      expo-cli: 2.18.2

Target: Expo Bare RN project 

### Steps to Reproduce
On Android devices:
1. Calling WebBrowser.openAuthSessionAsync(authorizeUri, redirectUri)
2. Gives back UnavailabilityError('WebBrowser', 'openBrowserAsync').

### Expected Behavior

Seems like Android doesn't have dismissBrowser(). From what I understand Android doesn't have dismissBrowser and it automatically dismiss the browser. However it is throwing me this error (I think when it shouldn't? since Android doesn't have the method so it shouldn't do anything.) 

### Reproducible Demo
https://github.com/kitty7756/testRepo

**Comment by cruzach** [comment]
Hi @kitty7756, can you clarify whether this is an Expokit project, or a bare RN project?

**Comment by kitty7756** [comment]
> Hi @kitty7756, can you clarify whether this is an Expokit project, or a bare RN project?

This is a bare RN project. (Sorry for the misleading description

**Comment by kitty7756** [comment]
@cruzach,
I have a follow up question regarding this issue:
I'm implementing an authentication flow in bare RN. Originally the authentication flow was written with Expo, and it was calling WebBrowser.openAuthSessionAsync(authorizeUri, redirectUri). What was it doing on Android device? Was it following the following flow: MainAppActivity -> Browser -> MainAppActivity? If it was, then wouldn't there be a security issue? I would like to know the answer since I was wondering if I have to wrap my Android folder to achieve the following flow: MainAppActivity -> AuthenticationActivity -> Browser -> RedirectActivity -> MainAppActivity.

**Comment by nicolazj** [comment]
I can confirm this issue also happens in Expokit with SDK33.
```
Error: The method or property WebBrowser.dismissBrowser is not available on android, are you sure you've linked all the native dependencies properly?
```

openBrowserAsync is availble in ExponentWebBrowser tho.

**Comment by bkonkle** [comment]
I'm hitting this issue in Expokit as well, with this code:

```
  const redirectUrl = AuthSession.getRedirectUrl()

  const result = await AuthSession.startAsync({
    authUrl: `${Config.Auth.domain}/authorize${toQueryString({
      ...(connection ? {connection} : {}),
      client_id: Config.Auth.clientId,
      response_type: 'token',
      scope: 'openid',
      audience: Config.Auth.audience,
      redirect_uri: redirectUrl,
    })}`,
  })
```

And this error:

```
[Unhandled promise rejection: Error: The method or property WebBrowser.dismissBrowser is not available on android, are you sure you've linked all the native dependencies properly?]
```

Using:

```
"expo": "^33.0.3",
"react-native": "https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz",
```

**Comment by changLiuUNSW** [comment]
Any updates? This issue blocks us from upgrading to expo 33.0.0.

**Comment by ide** [comment]
Published expo-web-browser 5.0.3 with a patch for auth session.

**Comment by sepehr-laal** [comment]
I am not sure if this was really patched, I am using `expo-web-browser: ^5.03` managed workflow, the browser just keeps hanging open after auth session ends on Android with the same error. Also tried the bare workflow with the same results.

**Comment by brentvatne** [comment]
@sepehr-laal - we would not have closed this and pointed out that it was fixed unless all of our tests showed that it was, but if you have found a case where this is not working then it would be most helpful for you to open an issue and provide the information requested in the issue template
